src/set.js

```diff
@@ -39,15 +39,38 @@ export default class Set {
     return this.set.length
   }
 
-union(otherSet) {
-  let unionSet = this.set
-  for(let item of otherSet) {
-    if(!unionSet.contains(item)) {
-      unionSet.append(item)
+  union(otherSet) {
+    let unionSet = this
+    for(let item of otherSet.set) {
+        unionSet.add(item)
     }
+    return unionSet
   }
-  return unionSet
-}
 
+  intersect(otherSet) {
+    let intersectSet = new Set([])
```

Comment:
Any reason for not using `const` here? My rule of thumb is to use `const` whenever I am not going to reassign the value of a variable, even though `const` does not enforce object/array immutability.